src/actions/sendUserOperation.ts

```diff
@@ -0,0 +1,116 @@
+import type { SmartWallet } from "../wallets/types";
+import { UserOperationStatusEnum } from "../client";
+import { Address, encodeFunctionData, Hex } from "viem";
+import { Coinbase } from "../coinbase/coinbase";
+import { CHAIN_ID_TO_NETWORK_ID, type SupportedChainId } from "../types/chain";
+import type { Call, Calls } from "viem/types/calls";
+import type { SendUserOperationParameters } from "viem/account-abstraction/";
+
+/**
+ * Options for sending a user operation
+ * @template T - Array type for the calls parameter
+ */
+export type SendUserOperationOptions<T extends readonly unknown[]> = {
+  /** Array of contract calls to execute in the user operation */
+  calls: Calls<T>;
+  /** Chain ID of the network to execute on */
+  chainId: SupportedChainId;
+  /** Optional URL of the paymaster service to use for gas sponsorship. Must be ERC-7677 compliant. */
+  paymasterUrl?: string;
+};
+
+/**
+ * Return type for the sendUserOperation function
+ */
+export type SendUserOperationReturnType = {
+  /** The ID of the user operation */
+  id: string;
+  /** The address of the smart wallet */
+  smartWalletAddress: Address;
+  /** The status of the user operation */
+  status: typeof UserOperationStatusEnum.Broadcast;
+};
+
+/**
+ * Sends a user operation to the network
+ *
+ * @example
+ * ```ts
+ * import { sendUserOperation } from "@coinbase/coinbase-sdk";
+ * import { parseEther } from "viem";
+ *
+ * const result = await sendUserOperation(wallet, {
+ *   calls: [
+ *     {
+ *       abi: erc20Abi,
+ *       functionName: "transfer",
+ *       args: [to, amount],
+ *     },
+ *     {
+ *       to: "0x1234567890123456789012345678901234567890",
+ *       data: "0x",
+ *       value: parseEther("0.0000005"),
+ *     },
+ *   ],
+ *   chainId: 1,
+ *   paymasterUrl: "https://api.developer.coinbase.com/rpc/v1/base/someapikey",
+ * });
+ * ```
+ *
+ * @param {SmartWallet} wallet - The smart wallet to send the user operation from
+ * @param {SendUserOperationOptions<T>} options - The options for the user operation
+ * @returns {Promise<SendUserOperationReturnType>} The result of the user operation
+ */
+export async function sendUserOperation<T extends readonly unknown[]>(
+  wallet: SmartWallet,
+  options: SendUserOperationOptions<T>,
+): Promise<SendUserOperationReturnType> {
+  const { calls, chainId, paymasterUrl } = options;
+  const network = CHAIN_ID_TO_NETWORK_ID[chainId];
+
+  const encodedCalls = calls.map(call => {
+    if ("abi" in call) {
+      return {
+        data: encodeFunctionData(call),
+        to: call.to,
+        value: call.value.toString(),
+      };
+    }
+    // return call as Call
+    return {
+      data: call.data,
+      to: call.to,
+      value: call.value.toString() || "0",
+    };
```

Comment:
If value is "undefined" then this returns an error. Can we do something like:

```suggestion
  const encodedCalls = calls.map(call => {
   const value = (call.value ?? BigInt(0)).toString()
  
    if ("abi" in call) {
      return {
        data: encodeFunctionData(call),
        to: call.to,
        value,
      };
    }
    // return call as Call
    return {
      data: call.data,
      to: call.to,
      value,
    };
```

or

```suggestion
        value: call.value != undefined ? value.toString() : "0"
```

Reply:
See latest code - should be fixed now. `value` is a BigInt, not a string, so i added appropriate default logic to set BigInt(0)